src/xg.hpp

```diff
@@ -99,6 +99,13 @@ class XG {
     size_t node_count;
     size_t edge_count;
     size_t path_count;
+    
+    // We need the w function, which we call the "where_to" function. It tells
+    // you, from a given visit at a given side, what visit offset if you go to
+    // another side.
+    int64_t where_to(int64_t current_side, int64_t visit_offset, int64_t new_side) const;
+    int64_t where_to(int64_t current_side, int64_t visit_offset, int64_t new_side,
+      vector<Edge>& edges, vector<Edge>& edges_out) const;
```

Comment:
This is since my functions make a lot (some times >100) calls to where_to where the current and new sides are the same (therefore the edges in and out are the same) but the only difference is the visit_offsets are different. In aggregate the calls to edges_on_end/edges_on_start were expensive so I took those calls out of where_to so that I could reuse them between calls.

This function isn't really useful outside this use case so I'll add a comment flagging it as such